### ISSUE TITLE
Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libaudiotags.so'

### ISSUE THREAD

**Issue by sanihaq** [erikas-taroza/audiotags#2]
```
build/linux/x64/debug/bundle/lib/libaudiotags.so: file too short
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
#2      api
ffi.dart:22
#3      api (package:audiotags/src/ffi.dart)
ffi.dart:1
#4      AudioTags.read
audiotags.dart:9
#5      _MyHomePageState.build.<anonymous closure>.<anonymous closure>
main.dart:90
#6      _InkResponseState.handleTap
ink_well.dart:1096
#7      GestureRecognizer.invokeCallback
recognizer.dart:253
#8      TapGestureRecognizer.handleTapUp
tap.dart:627
#9      BaseTapGestureRecognizer._checkUp
tap.dart:306
```

I'm on ubuntu 20.04

**Comment by erikas-taroza** [comment]
Sorry, the `CMakeLists.txt` for Linux had a typo in the download URL for the library. Published to [pub.dev](https://pub.dev/packages/audiotags) as v1.0.1.

**Comment by sanihaq** [comment]
still not fixed
new logs...
```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libaudiotags.so': /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /path-to-project/build/linux/x64/debug/bundle/lib/libaudiotags.so)
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
#2      api
ffi.dart:22
#3      api (package:audiotags/src/ffi.dart)
ffi.dart:1
#4      AudioTags.read
audiotags.dart:9
#5      _FilesListState.build.<anonymous closure>.<anonymous closure>
main.dart:123
#6      _InkResponseState.handleTap
ink_well.dart:1096
#7      GestureRecognizer.invokeCallback
recognizer.dart:253
#8      TapGestureRecognizer.handleTapUp
tap.dart:627
#9      BaseTapGestureRecognizer._checkUp
tap.dart:306

```

**Comment by erikas-taroza** [comment]
Fixed in v1.0.2

**Comment by sheikhhaziq** [comment]
Error is showing again in v1..1.3

**Comment by erikas-taroza** [comment]
> Error is showing again in v1..1.3

Can you create a new issue with the error you are getting?

**Comment by marcoxmediran** [comment]
> Can you create a new issue with the error you are getting?

This is on v1.1.3 and it's showing something like this... Are there any error logs I should be sending? 

```
E/flutter (23439): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libaudiotags.so': dlopen failed: library "libaudiotags.so" not found
E/flutter (23439): #0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
E/flutter (23439): #1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
E/flutter (23439): #2      api (package:audiotags/src/ffi.dart:22:22)
E/flutter (23439): #3      api (package:audiotags/src/ffi.dart)
E/flutter (23439): #4      AudioTags.write (package:audiotags/src/audiotags.dart:24:18)
E/flutter (23439): #5      Downloader.playground (package:ytm_downloader_gui/downloader.dart:44:21)
E/flutter (23439): <asynchronous suspension>
```

**Comment by erikas-taroza** [comment]
> > Can you create a new issue with the error you are getting?
> 
> This is on v1.1.3, building for Android, and it's showing something like this. Are there any error logs I should be sending?
> 
> ```
> E/flutter (23439): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'libaudiotags.so': dlopen failed: library "libaudiotags.so" not found
> E/flutter (23439): #0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
> E/flutter (23439): #1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
> E/flutter (23439): #2      api (package:audiotags/src/ffi.dart:22:22)
> E/flutter (23439): #3      api (package:audiotags/src/ffi.dart)
> E/flutter (23439): #4      AudioTags.write (package:audiotags/src/audiotags.dart:24:18)
> E/flutter (23439): #5      Downloader.playground (package:ytm_downloader_gui/downloader.dart:44:21)
> E/flutter (23439): <asynchronous suspension>
> ```

I have fixed this issue in 73657e0

**Comment by soumyaDghosh** [comment]
Hey @erikas-taroza facing this same issue with `1.4.0`

```bash
Building package executable... 
Built dartmat:dartmat.
Hello, Enter the audio to list its metadatas:
"/home/soumyaxubuntu/Music/#SundaySuspense | Tenida | Jhau Bongalow-r Rahasya Part 2 | Narayan Gangopadhyay | Mirchi Bangla.mp3"
Unhandled exception:
Invalid argument(s): Failed to load dynamic library 'libaudiotags.so': libaudiotags.so: cannot open shared object file: No such file or directory
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
#2      api (package:audiotags/src/ffi.dart:22:22)
#3      api (package:audiotags/src/ffi.dart)
#4      AudioTags.read (package:audiotags/src/audiotags.dart:20:20)
#5      AudioParser.getMeta (package:dartmat/audio.dart:16:28)
#6      main (file:///home/soumyaxubuntu/Documents/Dart/dartmat/bin/dartmat.dart:9:15)
#7      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:295:33)
#8      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```
